### PR TITLE
evaluate_vec in polynomial and reference in evaluate

### PR DESCRIPTION
**Pull request by pablodeymo**
evaluate_vec method added in polynomial.
evaluate receiving a reference

### CODE REVIEW

File: math/src/polynomial.rs [hunk 1]
```diff
@@ -52,13 +52,17 @@ impl<F: IsField> Polynomial<FieldElement<F>> {
...
 
-    pub fn evaluate(&self, x: FieldElement<F>) -> FieldElement<F> {
+    pub fn evaluate(&self, x: &FieldElement<F>) -> FieldElement<F> {
         self.coefficients
             .iter()
             .enumerate()
             .fold(FieldElement::zero(), |acc, (i, c)| acc + c * x.pow(i))
     }
 
+    pub fn evaluate_vec(&self, input: &[FieldElement<F>]) -> Vec<FieldElement<F>> {
```
Comment (Oppen) [inline math/src/polynomial.rs:62]:
```suggestion
    pub fn evaluate_slice(&self, input: &[FieldElement<F>]) -> Vec<FieldElement<F>> {
```

**Review by Oppen — APPROVED** [review]
Nit, but LGTM.

File: math/src/polynomial.rs (hunk 1 above)
Comment (pablodeymo) [reply math/src/polynomial.rs]: done!

**Review by Oppen — APPROVED** [review]

**Review by ilitteri — APPROVED** [review]
LGTM

**Review by ajgara — APPROVED** [review]

**Pull request merged**